applications/finetune-quora-embeddings/benchmark.py

```diff
@@ -0,0 +1,315 @@
+    from tqdm import tqdm
+    import numpy as np
+    import torch
+    import torch.nn as nn
+
+    dataset = load_from_disk(f"{DATASET_DIR}/{DATASET_NAME}")
+    test_dataset = dataset["test"]
+    co = Client(os.environ["COHERE_API_KEY"])
+    sem = asyncio.Semaphore(64)
+    sentences = []
+    sentences_id_to_embedding_mapping = {}
+    t1 = []
+    t2 = []
+    labels = []
+    for row in test_dataset:
+        s1, s2 = row["questions"]["text"]
+        id_1, id_2 = row["questions"]["id"]
+
+        if id_1 not in sentences_id_to_embedding_mapping:
+            sentences_id_to_embedding_mapping[id_1] = len(
+                sentences_id_to_embedding_mapping
+            )
+            sentences.append(s1)
+
+        if id_2 not in sentences_id_to_embedding_mapping:
+            sentences_id_to_embedding_mapping[id_2] = len(
+                sentences_id_to_embedding_mapping
+            )
+            sentences.append(s2)
+
+        t1.append(sentences_id_to_embedding_mapping[id_1])
+        t2.append(sentences_id_to_embedding_mapping[id_2])
+        labels.append(1 if row["is_duplicate"] else 0)
+
+    batch_size = 96
+    print(f"Extracted {len(sentences)} unique sentences")
+    tqdm_monitoring_bar = tqdm(total=len(sentences))
+
+    async def embed_text(texts, progress_bar: tqdm):
+        async with sem:
+            response = co.embed(
+                texts=texts,
+                model="embed-multilingual-v3.0",
+                input_type="clustering",
+            )
+            progress_bar.update(len(texts))
+            return response.embeddings
+
+    start = time.time()
+    coros = [
+        embed_text(sentences[start : start + batch_size], tqdm_monitoring_bar)
+        for start in range(0, len(sentences), batch_size)
+    ]
+    res = await asyncio.gather(*coros)
+    tqdm_monitoring_bar.close()
+    end = time.time()
+    print(f"Generated embeddings in {end-start}s")
+
+    # Flatten the list of lists and convert to tensor of floats
+    flattened_res = [item for sublist in res for item in sublist]
+    embeddings_tensor = torch.tensor(flattened_res).to("cuda")
+    embedding_layer = nn.Embedding.from_pretrained(embeddings_tensor)
+
+    t1_tensor = torch.as_tensor(t1, device="cuda")
+    e1 = embedding_layer(t1_tensor)
+
+    t2_tensor = torch.as_tensor(t2, device="cuda")
+    e2 = embedding_layer(t2_tensor)
+
+    cosine_scores = util.cos_sim(e1, e2)
+    predictions = np.diag(cosine_scores.cpu()).tolist()
+    return roc_auc_score(labels, predictions)
+
+
+@stub.local_entrypoint()
+def main():
+    from tabulate import tabulate
+
+    generate_dataset_split.remote()
+
+    res = {}
+    res["text-embeddings-ada-v2"] = benchmark_openai.remote()
+    res["embed-multilingual-v3.0"] = benchmark_cohere_roc.remote()
+
+    for model_name, auc in zip(
+        MODELS, benchmark_mteb_model.map(MODELS, order_outputs=True)
+    ):
+        res[model_name] = auc
+
+    values = [[model, auc] for model, auc in res.items()]
```

Comment:
```suggestion
    values = [(model, auc) for model, auc in res.items()]
```

don't hard code metric variables and name them metrics, we'll want `model, scores` at some point where scores = a suite of evals